drivers/nrf_radio_802154/platform/clock/nrf_802154_clock_zephyr.c

```diff
@@ -56,18 +56,24 @@ void nrf_802154_clock_deinit(void)
     /* Intentionally empty. */
 }
 
+static void hfclk_on_callback(struct device *dev, clock_control_subsys_t subsys,
+                              void *user_data)
+{
+    hfclk_is_running = true;
+    nrf_802154_clock_hfclk_ready();
+}
+
 void nrf_802154_clock_hfclk_start(void)
 {
+    static struct clock_control_async_data clk_data = {
+        .cb = hfclk_on_callback
+    };
     struct device *clk;
 
     clk = device_get_binding(DT_INST_0_NORDIC_NRF_CLOCK_LABEL);
```

Comment:
While we're on this topic, perhaps it would be good to also switch to using `DT_LABEL(DT_INST(0, nordic_nrf_clock))` here (and in 3 more places in this file). This would save us one update of this driver that we'd need in the near future.

Reply:
Thanks, updated. The Zephyr CI has been re-run.